feat(transformer/typescript): support `removeClassFieldsWithoutInitializer` option

Dunqing opened the pull request:
* close #9192 
* close #10491
 
We've discussed adding `removeClassFieldsWithoutInitializer` option to support removing class fields without an initializer in https://github.com/oxc-project/oxc/pull/10491#issuecomment-2823195571. This is used to align the`TypeScript`'s `useDefineForClassFields: false` option.

Dunqing commented:
@sapphi-red could you review this?

sapphi-red commented on crates/oxc_transformer/src/typescript/options.rs:
````diff
@@ -44,6 +44,42 @@ pub struct TypeScriptOptions {
 
+    /// When enabled, class fields without initializers are removed.
+    ///
+    /// For example:
+    /// ```ts
+    /// class Foo {
+    ///    x: number;
+    ///    y: number = 0;
+    /// }
+    /// ```
+    /// // transform into
+    /// ```js
+    /// class Foo {
+    ///    x: number;
+    ///    y: number = 0;
+    /// }
+    /// ```
````
The output of this example seems incorrect.

Dunqing replied:
Oh, sorry, my bad!

Dunqing committed:
feat(transformer/typescript): support `removeClassFieldsWithoutInitializer` option (#10576)

* close #9192
* close #10491

We've discussed adding `removeClassFieldsWithoutInitializer` option to support removing class fields without an initializer in https://github.com/oxc-project/oxc/pull/10491#issuecomment-2823195571. This is used to align the`TypeScript`'s `useDefineForClassFields: false` option.
